### PR TITLE
offline caching in home page

### PR DESCRIPTION
**Pull request by krishnachaitanya0107**

### CODE REVIEW

File: domain/src/main/java/divyansh/tech/animeclassroom/home/HomeDefaultRepo.kt [hunk 1]
```diff
@@ -2,11 +2,16 @@ package divyansh.tech.animeclassroom.home
 
 import android.util.Log
 import divyansh.tech.animeclassroom.ResultWrapper
-import divyansh.tech.animeclassroom.api.HomeScreenApi
+import divyansh.tech.animeclassroom.models.home.AnimeModel
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.flow
 import javax.inject.Inject
 
+const val POPULAR_ANIME="PopularAnime"
```
Comment (justdvnsh) [inline domain/src/main/java/divyansh/tech/animeclassroom/home/HomeDefaultRepo.kt]: The tags are already made in an enum class. If you want to save them to the database, use enum classes ordinal. 
```
enum class HOME_TYPES(value: Int) {
    POPULAR_ANIME(1),
    RECENT_RELEASE(2)
    ....// and so on. 
}
```

Also, to use the values we can just use HOME_TYPES.POPULAR_ANIME.ordinal or something like this.

File: domain/src/main/java/divyansh/tech/animeclassroom/home/HomeDefaultRepo.kt [hunk 2]
```diff
@@ -23,6 +28,9 @@ class HomeDefaultRepo @Inject constructor(
         //TODO: Add local data first. (caching purpose)
         return flow {
```
Comment (justdvnsh) [inline domain/src/main/java/divyansh/tech/animeclassroom/home/HomeDefaultRepo.kt:25]: Here it should be fetching the animes from the local database only first and then if there is no data in cache, then we would be making the network request.

File: domain/src/main/java/divyansh/tech/animeclassroom/home/HomeDefaultRepo.kt [hunk 3]
```diff
@@ -44,9 +58,14 @@ class HomeDefaultRepo @Inject constructor(
     * @returns: Flow<ResultWrapper<*>>
     * */
     suspend fun parsePopularMovies(): Flow<ResultWrapper<*>> {
-        //TODO: Add local data first. (caching purpose)
         return flow {
             val response = remoteRepo.getPopularMovies()
```
Comment (justdvnsh) [inline domain/src/main/java/divyansh/tech/animeclassroom/home/HomeDefaultRepo.kt]: same

File: domain/src/main/java/divyansh/tech/animeclassroom/home/HomeDefaultRepo.kt [hunk 4]
```diff
@@ -56,9 +75,14 @@ class HomeDefaultRepo @Inject constructor(
     * @returns: Flow<ResultWrapper<*>>
     * */
     suspend fun parseNewSeasons(): Flow<ResultWrapper<*>> {
-        //TODO: Add local data first. (caching purpose)
         return flow {
             val response = remoteRepo.getNewSeasons()
```
Comment (justdvnsh) [inline domain/src/main/java/divyansh/tech/animeclassroom/home/HomeDefaultRepo.kt]: same

**Review by justdvnsh — CHANGES_REQUESTED** [review]
Hi @krishnachaitanya0107 . I have reviewed the files. Please make the required changes

**Comment by krishnachaitanya0107** [comment]
@justdvnsh  i implemented the changes , please have a look and let me know if anything can be changed  .

one thing i wish to mention - I had to move utils/Hometypes from app module to utils module because i was facing a weird 'circular dependency between the following tasks' issue  .

**Comment by justdvnsh** [comment]
@krishnachaitanya0107 Hi, can you resolve the merge conflicts and also, can you please provide a working video of your feature. Thanks

**Comment by krishnachaitanya0107** [comment]
Hey , resolved the merge conflicts . About the working video , there isn't any difference that i can show in the screen recording because this feature doesn't have any ui related changes . I actually used logs to see whether data is coming from room database while testing .

**Comment by krishnachaitanya0107** [comment]
@justdvnsh , please do let me know if you want any more changes in the code . I will be happy to implement them .

**Review by justdvnsh — APPROVED** [review]
Looks good to me. Thank your for your contributions.

**Comment by krishnachaitanya0107** [comment]
Thank you @justdvnsh for going through my code , It's amazing to know that my contributions are helping out .

**Comment by justdvnsh** [comment]
@krishnachaitanya0107 Offline cache has been added, but it has disrupted the flow of favorites. See the favorites tab. It will show you all the cached items. We don't want that. I am making a new issue, you can take that up if you want. The solution is pretty simple. You just need to create a separate database for the cache rather than the usual database.

**Comment by krishnachaitanya0107** [comment]
Oops , Really sorry @justdvnsh  i did not mean to mess up that flow 😅 . Ok i will take up the new issue . It would be great if you could guide me a bit on this .

**Pull request merged**